server/src/test/java/org/elasticsearch/repositories/RepositoryDataTests.java

```diff
@@ -110,7 +111,7 @@ public void testAddSnapshots() {
             newSnapshot,
             new RepositoryData.SnapshotDetails(
                 randomFrom(SnapshotState.SUCCESS, SnapshotState.PARTIAL, SnapshotState.FAILED),
-                randomFrom(Version.CURRENT, Version.CURRENT.minimumCompatibilityVersion()),
+                randomFrom(IndexVersion.CURRENT, Version.CURRENT.minimumCompatibilityVersion().indexVersion),
```

Comment:
Does this actually need to be minimumCompatibilityVersion, or should it be minimumIndexCompatibilityVersion?

Reply:
What I understand for Snapshot restore is that we don't want to restore an incompatible index into the cluster, so it's the index version we care about. But is the repository data format defined by the transport version? I'm not sure.

Reply:
I'll be looking at the repository version specifically in a separate PR

Reply:
We can leave this as it is for now, and re-evaluate as part of the final push to remove the `indexVersion` field